creator/html_paragraph.go

```diff
@@ -0,0 +1,625 @@
+		w = math.Max(w, e.Width())
+	}
+	return w
+}
+
+// Height returns the height of the Drawable.
+func (b *htmlBlock) Height() float64 {
+	var h float64
+	for _, e := range b.elements {
+		h = math.Max(h, e.Height())
+	}
+	return h
+}
+
+func (b *htmlBlock) SetWidth(w float64) {
+	b.style.getOrCreateElementStyle().width = &w
+}
+
+// GeneratePageBlocks generates the page blocks.  Multiple blocks are generated if the contents wrap
+// over multiple pages. Implements the Drawable interface.
+func (b *htmlBlock) GeneratePageBlocks(ctx DrawContext) ([]*Block, DrawContext, error) {
+	var blocks []*Block
+
+	if es := b.style.ElementStyle; es != nil {
+		blockCtx := ctx
+		blockCtx.Height = 0.0
+		for _, e := range b.elements {
+			blockCtx.Height += e.Height()
+		}
+
+		block := NewBlock(blockCtx.PageWidth, blockCtx.PageHeight)
+		block.xPos = ctx.X
+		block.yPos = ctx.Y
+		blocks = append(blocks, block)
+		border := newBorder(blockCtx.X, blockCtx.Y, blockCtx.Width, blockCtx.Height)
+
+		if es.backgroundColor != nil {
+			r := es.backgroundColor.R()
+			g := es.backgroundColor.G()
+			b := es.backgroundColor.B()
+
+			border.SetFillColor(ColorRGBFromArithmetic(r, g, b))
+		}
+
+		border.LineStyle = es.borderLineStyle
+
+		border.styleLeft = es.borderStyleLeft
+		border.styleRight = es.borderStyleRight
+		border.styleTop = es.borderStyleTop
+		border.styleBottom = es.borderStyleBottom
+
+		if es.borderColorLeft != nil {
+			border.SetColorLeft(ColorRGBFromArithmetic(es.borderColorLeft.R(), es.borderColorLeft.G(), es.borderColorLeft.B()))
+		}
+		if es.borderColorBottom != nil {
+			border.SetColorBottom(ColorRGBFromArithmetic(es.borderColorBottom.R(), es.borderColorBottom.G(), es.borderColorBottom.B()))
+		}
+		if es.borderColorRight != nil {
+			border.SetColorRight(ColorRGBFromArithmetic(es.borderColorRight.R(), es.borderColorRight.G(), es.borderColorRight.B()))
+		}
+		if es.borderColorTop != nil {
+			border.SetColorTop(ColorRGBFromArithmetic(es.borderColorTop.R(), es.borderColorTop.G(), es.borderColorTop.B()))
+		}
+
+		border.SetWidthBottom(es.borderWidthBottom)
+		border.SetWidthLeft(es.borderWidthLeft)
+		border.SetWidthRight(es.borderWidthRight)
+		border.SetWidthTop(es.borderWidthTop)
+
+		err := block.Draw(border)
+		if err != nil {
+			common.Log.Debug("ERROR: %v", err)
+		}
+	}
+
+	for _, e := range b.elements {
+		var newBlocks []*Block
+		var err error
+		newBlocks, ctx, err = e.GeneratePageBlocks(ctx)
+		if err != nil {
+			return nil, ctx, err
+		}
+		if len(newBlocks) < 1 {
+			continue
+		}
+		if len(blocks) == 0 {
+			blocks = newBlocks[0:1]
+		} else {
+			blocks[len(blocks)-1].mergeBlocks(newBlocks[0])
+		}
+
+		blocks = append(blocks, newBlocks[1:]...)
+	}
+	return blocks, ctx, nil
+}
+
+// NewHTMLParagraph creates a new html paragraph.
+// Default attributes:
+// Font: Helvetica,
+// Font size: 10
+// Encoding: WinAnsiEncoding
+// Wrap: enabled
+// Text color: black
+func (c *Creator) NewHTMLParagraph() *HTMLParagraph {
```

Comment:
Please move `NewHTMLParagraph` in the `creator.go` file with all the other component creation methods.